libs/util/json/src/lib/util-json.ts

```diff
@@ -5,3 +5,14 @@ export const isValidJSON = (str: string) => {
     return false;
   }
 };
+
+export const JsonFormatting = (json: string) => {
+  if (Object.prototype.toString.call(json) === "[object Object]") {
+    json = JSON.stringify(json);
+  }
+  const isValid = isValidJSON(json);
+  if (isValid) {
+    return JSON.stringify(JSON.parse(json!), null, "\t");
+  }
+  return json;
+};
```

Comment:
@Tjsingh01996 avoid using json parse and stringify multiple times since it is an heavy load operation. 

```suggestion
export  const JsonFormatting = (json: any) => {
  if (Object.prototype.toString.call(json) === "[object Object]") {
    json = JSON.stringify(json);
  }
  const isValid = isValidJSON(json);
  if (isValid) {
    return JSON.stringify(data, null, 2);
  }
  return json;
};
```

@Tjsingh01996 can you add unit tests for this new method?

Reply:
@overbit 
const jsonParse = (str: string) => {
  try {
    let data = JSON.parse(str)
    return data && data;
  } catch (e) {
    return false;
  }
}

export const JsonFormatting = (json: any) => {
  let parsedJson;
  if (Object.prototype.toString.call(json) === "[object Object]") {
    json = JSON.stringify(json);
  }
  parsedJson = jsonParse(json);
  if (parsedJson) {
    return JSON.stringify(parsedJson, null, 2);
  }
  return json;
};

Reply:
Sorry @Tjsingh01996 I don't understand your comment.  IsValidJson and jsonParse use the same JSON.parse. How is this different from the first implementation?

Reply:
can you give me suggestion how i can do this 
isValid was validating the json and return. boolean 
but in jsonParse function. i am returning  parsed data if it validate 
in this way we can use json.parse single time

Reply:
export const isValidJSON = (str: string) => {
  try {
    return JSON.parse(str) && !!str;
  } catch (e) {
    return false;
  }
};

Reply:
@overbit 
export const JsonFormatting = (data: any) => {
  if (typeof data !== "string") {
    return JSON.stringify(data, null, 2);
  }
  try {
    let object = JSON.parse(data);
    return JSON.stringify(object, null, 2);
  } catch (e) {
    return data
  }
};

Reply:
can we do like this ?
this code also support the string if we pass in the fuction